src/makielayout/blocks.jl

```diff
@@ -16,6 +16,7 @@ macro Block(name::Symbol, body::Expr = Expr(:block))
             parent::Union{Figure, Scene, Nothing}
             layoutobservables::Makie.LayoutObservables{GridLayout}
             blockscene::Scene
+            finalizers::Vector{Any}
```

Comment:
In JSServe, I started having these signatures:
```
map(f, session, args...)
on(f, session, obs)
onany(f, session, obs)
```
I was wondering if we shouldn't have this as the main API to bind observables to the lifecycle of another object?
Not sure if we can make it an official Observables API, but i think we should definitely add:
```julia
map(f, scenelike, args...)
on(f, scenelike, obs)
onany(f, scenelike, obs)
```
And use that in place where you now have the `x = on(...); push!(ax.finalizers, x)`

Reply:
That depends on what you mean by `lifecycle`, because I don't think it should mean garbage collection. With all our cross-connections between objects, it's likely that a `delete!`d `Axis` for example does not get garbage collected for a long time, if ever. The behavior when calling `delete!` should be entirely predictable.